scadalts-ui/src/views/DataObjects/Scripting/index.vue

```diff
@@ -1,33 +1,401 @@
 <template>
 	<div>
-		<iframe
-			class="scada-classic-frame"
-			width="100%"
-			height="100%"
-			frameborder="0"
-			:src="location"
-		>
-		</iframe>
+		<h1>{{ $t('scriptList.title') }}</h1>
+
+		<v-container fluid v-if="!!scriptList">
+			<v-card>
+				<v-card-title>
+					<v-text-field
+						v-model="search"
+						@input="fetchScriptList"
+						append-icon="mdi-magnify"
+						label="Search"
+						class="mr-2"
+						single-line
+						hide-details
+					></v-text-field>
+					<v-btn color="primary" @click="createNewScript">{{
+						$t('scriptList.newScript')
+					}}</v-btn>
+				</v-card-title>
+				<v-data-table
+					:headers="headers"
+					:items="scriptListFiltered"
+					:options.sync="options"
+					:server-items-length="totalScripts"
+					multi-sort
+					class="elevation-1"
+					@click:row="selectScript($event.id)"
+				>
+					<template v-slot:item.actions="{ item }">
+						<v-icon class="mr-2" border="0" @click.stop="runScript(item.xid)" title="run">
+							mdi-cog
+						</v-icon>
+						<v-icon border="0" @click.stop="deleteScript(item.id)" title="delete">
+							mdi-delete
+						</v-icon>
+					</template>
+				</v-data-table>
+			</v-card>
+		</v-container>
+
+		<v-row justify="center">
+			<v-dialog
+				v-model="dialog"
+				v-show="selectedScriptId !== null"
+				@change="selectedScriptId = null"
+			>
+				<v-card>
+					<v-card-title>
+						<v-row>
+							<v-col v-if="selectedScriptId != -1" cols="6">Script #{{ selectedScriptId }}</v-col>
+							<v-col v-else cols="6">{{ $t('scriptList.newScript') }}</v-col>
+							<v-col cols="12">
+								<v-btn
+									v-if="selectedScriptId != -1"
+									class="mr-2"
+									color="blue"
+									@click="runScript(selectedScript.xid)"
+								>
+									<v-icon>mdi-cog</v-icon>
+									{{ $t('scriptList.run') }}
+								</v-btn>
+
+								<v-btn
+									v-if="selectedScriptId != -1"
+									@click="deleteScript(selectedScriptId)"
+									class="mr-2"
+									color="red"
+								>
+									<v-icon>mdi-delete</v-icon>
+									{{ $t('scriptList.delete') }}
+								</v-btn>
+							</v-col>
+						</v-row>
+					</v-card-title>
+					<v-card-text>
+						<form>
+							<v-row>
+								<v-col cols="6">
+									<v-text-field ref="xidInput" :label="$t('common.xid')" @input="validateXid"  v-model="scriptForm.xid" :rules="[ruleNotNull, ruleXidUnique, ruleMaxLen50 ]"></v-text-field>
+								</v-ruleXidUniquValidScriptBodyext-field>
+								</v-col>
+								<v-col cols="6">
+									<v-text-field :label="$t('common.name')" v-model="scriptForm.name" :rules="[ruleNotNull, ruleMaxLen40 ]"></v-text-field>
+								</v-col>
+								<v-col cols="6">
+									<v-select
+										item-value="id"
+										placeholder="select datapoint"
```

Comment:
This placeholder 'literal' can be replaced by the i18n object. Or to be consistent use upperCase for the first letter because rest of the inputs fields properties starts from the upper case letter.

Reply:
fixed

Reply:
fixed

Reply:
Great! But you fixed this only in one place. You should try to replace all that 'literals' occurrences in this file.

Reply:
Great! But you fixed this only here. Try to replace all 'literals' with the i18n object in this file.